packages/types/src/clerk.ts

```diff
@@ -730,6 +730,16 @@ export type UserButtonProps = {
   userProfileProps?: Pick<UserProfileProps, 'additionalOAuthScopes'>;
 };
 
+type PrimitiveKeys<T> = {
+  [K in keyof T]: T[K] extends string | boolean | number | null ? K : never;
+}[keyof T];
+
+type PrimitiveOrganizationResource = PrimitiveKeys<OrganizationResource>;
+type PrimitiveUserResource = PrimitiveKeys<UserResource>;
+
+type OrganizationURL = `:${PrimitiveOrganizationResource}` | (string & {});
```

Comment:
Can we see a few examples of how IDE intellisense would look here?

Reply:
Also, since we don't export these, we can probably create a generic type instead of creating `OrganizationURL` and `PersonalWorkspaceURL` separately and use that as the return value below